src/ape/managers/project/types.py

```diff
@@ -90,30 +90,34 @@ def create_manifest(
 
             cached_sources = manifest.sources or {}
             cached_contract_types = manifest.contract_types or {}
-            sources = (
-                {s for s in self.sources if s in file_paths} if file_paths else set(self.sources)
+            cached_source_references = {
+                source_id: getattr(source, "references", [])
+                for source_id, source in cached_sources.items()
+            }
+            source_paths = (
+                {p for p in self.sources if p in file_paths} if file_paths else set(self.sources)
             )
 
-            # Filter out deleted sources
+            # Filter out deleted source_paths
             deleted_source_ids = cached_sources.keys() - set(
-                map(str, [get_relative_path(s, self.contracts_folder) for s in sources])
+                map(str, [get_relative_path(p, self.contracts_folder) for p in source_paths])
             )
             contract_types = {
                 name: contract_type
                 for name, contract_type in cached_contract_types.items()
                 if contract_type.source_id not in deleted_source_ids
             }
 
-            def file_needs_compiling(source: Path) -> bool:
-                path = str(get_relative_path(source, self.contracts_folder))
+            def source_ids_need_compiling(source_path: Path) -> bool:
```

Comment:
```suggestion
            def need_compiling(source_path: Path) -> bool:
```

I think the rename here was by accident anyway!

Reply:
Updated [b8b78ed](https://github.com/ApeWorX/ape/pull/730/commits/b8b78ed1cdf6910721bd7cb4b168dfae1bc1f1f4)